packages/zod-openapi/src/export.ts

```diff
@@ -0,0 +1,72 @@
+import { stringify as yamlStringify } from 'yaml'
+import type { OpenAPIHono } from './index'
+
+export interface ToFilesOptions {
+  outDir?: string
+  format?: 'json' | 'yaml'
+  extension?: string
+}
+
+export interface FileSystem {
+  writeFile(path: string, data: string | Uint8Array): Promise<void>
+  mkdir(path: string, options: { recursive: boolean }): Promise<void | string>
+}
+
+export interface ToFilesResult {
+  success: boolean
+  outDir: string
+  files: string[]
+  error?: Error
+}
+
+const generateFilePath = (basePath: string, docEndpointPath: string, format: string, extension?: string): string => {
+  const ext = extension || (format === 'yaml' ? '.yaml' : '.json')
+  const sanitizedPath = docEndpointPath.replace(/\//g, '-').replace(/^-*|-*$/g, '')
+  return `${basePath}/openapi-${sanitizedPath}${ext}`
+}
+
+const fetchAndSaveOpenAPIDocument = async (app: OpenAPIHono, fs: FileSystem, docEndpointPath: string, filePath: string, format: string): Promise<string> => {
+  const response = await app.request(`http://localhost${docEndpointPath}`)
```

Comment:
IIUC you don't have to add a doc endpoint in the first place, because you can obtain the OpenAPI document via `getOpenAPIDocument`. Am I missing something?